jmxtrans-core/src/main/java/com/googlecode/jmxtrans/model/Server.java

```diff
@@ -306,6 +317,17 @@ private Server(
 			environment.put(JMXConnector.CREDENTIALS, credentials);
 		}
 
+		if (ssl) {
```

Comment:
An integration test for this would be really welcomed! But it is probably non-trivial to write...